src/state/types.ts

```diff
@@ -44,20 +44,41 @@ export type LazyConfigFeatureList = {
   [key in LazyConfigFeaturesNames]?: ConfigFeature;
 };
 
-export type ExtResource = {
-  general: {
-    resource: {
-      kind: string;
-      group: string;
-      version: string;
-    };
-    name: string;
-    category: string;
-    urlPath: string;
-    scope: 'namespace' | 'cluster';
-    description?: string;
-    icon?: IconGlyph;
+export type ExtWidget = {
+  name: string;
+  widget: string;
+  source: string;
+  slot: 'top' | 'bottom';
+  destination: 'ClusterOverview' | 'Pods';
```

Comment:
do we want to limit the choices in the type? especially when extensibility resources kick in?
i think both of these should simple be `string`